.github/workflows/pr_validate.yaml

```diff
@@ -0,0 +1,193 @@
+      sandbox: ${{ steps.extract-staging.outputs.sandbox }}
+      is-e2e-box-checked: ${{ steps.check-e2e-checkbox.outcome }}
+    steps:
+      - id: install
+        name: Install requirements
+        run: |-
+          sudo apt-get update -y
+          sudo apt-get install -y pcregrep
+
+      - id: check-issue-url
+        name: Check issue URL
+        run: |-
+          REGEX="#### URL to issue\s?\r\n(https:\/\/biomage\.atlassian\.net\/browse\/BIOMAGE-[0-9]+|N\/A)"
+
+          echo "REGEX to test against:"
+          echo $REGEX
+          echo "Now looking at regex result:"
+
+          pcregrep -o1 -M "$REGEX" <<\EOF
+          ${{ github.event.pull_request.body }}
+          EOF
+
+      - id: extract-staging
+        name: Check staging URL
+        run: |-
+          REGEX="#### Link to staging deployment URL \(or set N\/A\)\s?\r\n(https:\/\/ui-(.+)\.scp-staging\.biomage\.net|N\/A)"
+
+          echo "REGEX to test against:"
+          echo $REGEX
+          echo "Now looking at regex result:"
+
+          URL=$(pcregrep -o1 -M "$REGEX" <<\EOF
+          ${{ github.event.pull_request.body }}
+          EOF
+          )
+
+          IS_STAGING_NA='false'
+
+          if [ $URL = 'N/A' ]; then
+            IS_STAGING_NA='true'
+          fi
+
+          echo "Is staging N/A?"
+          echo $IS_STAGING_NA
+          echo "::set-output name=is-staging-na::$IS_STAGING_NA"
+
+          echo "Full URL:"
+          echo $URL
+          echo "::set-output name=url::$URL"
+
+          SANDBOX=$(pcregrep -o2 -M "$REGEX" <<\EOF
+          ${{ github.event.pull_request.body }}
+          EOF
+          )
+          echo "Extracted sandbox:"
+          echo $SANDBOX
+          echo "::set-output name=sandbox::$SANDBOX"
+
+      - id: reach-staging
+        if: steps.extract-staging.outputs.url != 'N/A'
+        name: Attempt to reach staging environments
+        uses: nick-invision/retry@v2
+        with:
+          timeout_seconds: 30
+          max_attempts: 5
+          retry_on: error
+          command: curl -f ${{ steps.extract-staging.outputs.url }}
+          # Add jitter to break up correlated events.
+          on_retry_command: sleep $[($RANDOM % 10) + 5]s
+
+      - id: check-e2e-checkbox
+        name: Check if E2E testing is checked
+        continue-on-error: true
+        run: |-
+          REGEX="\[[Xx]\] Started end-to-end tests on the latest commit"
+    
+          pcregrep -o1 -M "$REGEX" <<\EOF
+          ${{ github.event.pull_request.body }}
+          EOF
+
+  run-integration-tests:
+    name: Run integration tests
+    runs-on: ubuntu-20.04
+    needs: lint-pr-body
+    concurrency: 
+      group: ${{ github.sha }}
+    steps:
+      - id: cancel
+        name: Cancel if staging is not N/A and box is not checked
+        if: needs.lint-pr-body.outputs.is-staging-na == 'false' && needs.lint-pr-body.outputs.is-e2e-box-checked != 'success'
+        uses: andymckay/cancel-action@0.2
```

Comment:
We have to run integration tests even if the box is not checked because a skipped job does not equal to a failed / cancelled one and we can not use it as a status check.